src/v/config/bounded_property.h

```diff
@@ -199,11 +269,9 @@ class bounded_property : public property<T> {
 
         if (_bounds.min.has_value() && _bounds.max.has_value()) {
             // Take midpoint of min/max and align it.
-            guess = _bounds.min.value()
-                    + (_bounds.max.value() - _bounds.min.value()) / 2;
-            if (_bounds.align.has_value()) {
-                guess -= guess % _bounds.align.value();
-            }
```

Comment:
align is done in `_bounds.clamp()` now. `numeric_integral_bounds::clamp()` does both clamp and align. Actual clamping is not needed in the call below since the value is the median, but the align part of the clamp is what it is there for.

Maybe `clamp` is not a good name for it and should be changed. In general, this function applies all the restriction laid by the Bounds class and produces a valid value as close to the argument as possible. Maybe `adjust`?

Reply:
I think it's all good. thanks for the clarification.